pydantic/main.py

```diff
@@ -487,17 +511,56 @@ def _decompose_class(cls: Type['Model'], obj: Any) -> GetterDict:
         return GetterDict(obj)
 
     @classmethod
-    def _get_value(cls, v: Any, by_alias: bool, skip_defaults: bool) -> Any:
+    @no_type_check
```

Comment:
It will be on every line we call this function, so I thought having decorator would be better. But yeah, I can replace it with `# type: ignore` if you wish

Reply:
no that's. Sounds like I need to look at this more.

Reply:
MyPy throws error here because we use `and` operator on `exclude=value_items and value_items.for_value(k)`  
```py
pydantic/main.py:597: error: Argument "exclude" to "_get_value" of "BaseModel" has incompatible type "Union[ValueItems, None, Any]"; expected "Union[Set[Union[int, str]], Dict[Union[int, str], Any], None]"
```
So it thinks that potentially we can pass an instance of `ValueItems` itseld, but since `bool(ValueItems())` always will be equivalent for `True`, it will never happen, so I guess we absolutely ok here.

Reply:
> - it doesn't slow things down

I guess it definitely will slow something down. Unfortunately I didn't run any comparison benchmarks. So please let me know what you find out

> - the whole _iter vs. _get_values thing looks a bit ugly ...

Agree on that and will appreciate any suggestions how we can make this look better

> please could you rebase and update HISTORY ...

@samuelcolvin yep, done.